Formula/sip.rb

```diff
@@ -31,14 +31,23 @@ def install
     end
 
     Language::Python.each_python(build) do |python, version|
-      # Note the binary `sip` is the same for python 2.x and 3.x
+      # Avoid picking up /Application/Xcode.app paths
+      ENV.delete("SDKROOT")
       system python, "configure.py",
                      "--deployment-target=#{MacOS.version}",
                      "--destdir=#{lib}/python#{version}/site-packages",
                      "--bindir=#{bin}",
                      "--incdir=#{include}",
                      "--sipdir=#{HOMEBREW_PREFIX}/share/sip"
       system "make"
+      # Use opt_prefix instead of keg path with brewed pythons
+      if version == "2.7" && Formula["python"].installed?
```

Comment:
maybe check the location of `python`?

Reply:
What are we doing here if the user isn't using Homebrew's `python`? The Requirement searches the whole `$PATH` right, so if building from source they could plausibly want something that is neither brew nor system `python`, no?

Reply:
I believe the requirement currently strictly prefers Homebrew's Python if it's installed.

Reply:
But presumably if it isn't this could cause some unexpected behaviour? I mean I guess you/Mike may argue that Homebrew is essentially not a from-source package manager any more, so maybe the discussion point is effectively nullified by that.

Reply:
I don't understand the point. If `Formula["python"].installed?` is true and the require always uses the formula if it's installed, then what are you asking?

Reply:
Oh, crap. I can't read. Sorry, I thought an earlier version of this PR was hardcoding in `/System` in the event Homebrew's `python` wasn't found & so the choice was essentially reduced to either system or brew & nothing else, even when building from source, and I guess I got hung up on that :see_no_evil:.

Forgive me, I slept for 3 hours before someone two floors above filled their kitchen with smoke & forced a 45 minute evacuation this morning.

Reply:
Sounds fun.

Of course, this could just use `inreplace "file", "foo", "bar", false` and call it a day.

Reply:
Or fixing the `python-config` scripts & retaining just the `ENV.delete("SDKROOT")` here, presumably? I don't know. Maybe I should stop making suggestions for the day :laughing:.

Reply:
>Or "fixing" the python-config scripts

Yes, I suggested that to apjanke yesterday.

Reply:
We do the same thing in a bunch of places; I think it'd be a reasonable step, personally.